conda_smithy/templates/build_steps.sh.tmpl

```diff
@@ -27,6 +27,22 @@ conda-build:
  root-dir: ${FEEDSTOCK_ROOT}/build_artifacts
 
 CONDARC
+
+if [ ${SINGULARITY_BUILD} == "True" ]; then
+    ### rough workaround to avoid singularity
+    ### read-only defaults:
```

Comment:
what does this mean?

Reply:
Singularity, by default, only does read-only when accessing images. (Otherwise, one would need to sandbox the image, and it gets pretty complicated to implement that). In the case of conda-forge docker images, the problem is that this line (just a few lines down from this statement)

```
mamba install --update-specs --yes --quiet --channel conda-forge \
    conda-build pip boa conda-forge-ci-setup=3
```

will basically need to edit a read-only filesystem (located in /opt/conda), and thus it fails. The workaround is to create a new environment from scratch which will be in the user space and thus editable, where the `mamba install --...` line will proceed successfully.

Does this make sense?

Reply:
Even with this implementation, I still get this nonfatal error about the filesystem:

```
error    libmamba Error opening for writing "/opt/conda/pkgs/urls.txt": Permission denied
```

It could be worth restructuring our images so that they're always in the user space with no need whatsoever for accessing places like /opt/, etc.

Reply:
How is this fixed?

Reply:
`error    libmamba Error opening for writing "/opt/conda/pkgs/urls.txt": Permission denied` is not important (it is nonfatal)

Reply:
For more context on this error message, Issue https://github.com/mamba-org/mamba/issues/1697 describes the source of the error message, confirms that it is harmless, and also notes that it will be removed in future versions of mamba to avoid confusion

Reply:
@jdblischak thanks for linking this!

Btw, this PR here is to enable HPC users to build on singularity instead of docker, thus allowing people with access to HPCs to help build and test complicated packages like tensorflow and pytorch. You're welcome to try it and let us know if you think a more productive approach is possible... currently it is a bit hacky and tortuous, but it works.